### PR TITLE
Add a help form page type for donors to contact us

### PR DESCRIPTION
**Pull request by Pomax**
This adds a new model and associated template + css to set up donor help pages that forward into Salesforce, using whatever orgid is set as `SALESFORCE_ORGID` in the instance's environment variables.

It's rather bare at the moment, with just enough CSS and markup to make things quite easy to turn into a design that works well enough.

Demo url: https://donate-wagtail-staging-pr-696.herokuapp.com/en-US/help/
Demo redirect url: https://donate-wagtail-staging-pr-696.herokuapp.com/en-US/help/?submitted=true

Note that the slug is derived from the title, but it doesn't need to be in the slightest and we can give the page whatever title we like, then set a custom `help` slug.

### CODE REVIEW

File: donate/templates/pages/core/contributor_support_page.html [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        <h1 class="page-header__heading">{{ page.title }}</h1>
+    </div>
+</header>
+
+<div class="layout">
+    <div class="layout__container">
+        <div class="layout__full-col">
+            <div class="page-body">
+                <p>{% blocktrans %}
+                    Here is some intro guide text to explain what this form is for and who should be filling it out.
```
Comment (Pomax) [inline donate/templates/pages/core/contributor_support_page.html]: We probably want a better text here =)

File: donate/templates/pages/core/contributor_support_page.html [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                        <label for="subject">Subject</label>
+                        <input id="subject" maxlength="80" name="subject" size="20" type="text">
+                    </fieldset>
+
+                    <fieldset>
+                        <label for="description">Description</label>
+                        <textarea name="description"></textarea>
+                    </fieldset>
+
+                    <input type="submit" name="submit">
```
Comment (Pomax) [inline donate/templates/pages/core/contributor_support_page.html]: "submit" is kind of a shitty word, what's a nicer one?

File: donate/settings.py [hunk 3]
```diff
@@ -536,9 +538,10 @@
         'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
     },
 ]
-AUTHENTICATION_BACKENDS = (
-    'mozilla_django_oidc.auth.OIDCAuthenticationBackend',
-)
+
+# AUTHENTICATION_BACKENDS = (
+#     'mozilla_django_oidc.auth.OIDCAuthenticationBackend',
+# )
```
Comment (alanmoo) [inline donate/settings.py]: Is this just for review?

File: donate/settings.py [hunk 4]
```diff
@@ -575,6 +578,9 @@
 STRIPE_API_KEY = env('STRIPE_API_KEY')
 STRIPE_WEBHOOK_SECRET = env('STRIPE_WEBHOOK_SECRET')
 
+# Salesforce
+SALESFORCE_ORGID = env('SALESFORCE_ORGID')
```
Comment (alanmoo) [inline donate/settings.py]: Does this need to be an env var or can we hardcode it, with an override in the Thunderbird app?

File: donate/templates/pages/core/contributor_support_page.html [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                        <label for="subject">{% trans "Subject" %}</label>
+                        <input id="subject" maxlength="80" name="subject" size="20" type="text">
+                    </fieldset>
+
+                    <fieldset>
+                        <label for="description">{% trans "Description" %}</label>
+                        <textarea name="description"></textarea>
+                    </fieldset>
+
+                    <input type="submit" name="submit" value="{% trans "Submit" %}">
```
Comment (Pomax) [inline donate/templates/pages/core/contributor_support_page.html]: "submit" is kind of a shitty word, what's a nicer one?

File: donate/templates/pages/core/contributor_support_page.html [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+                    </fieldset>
+
+                    <fieldset class="form-item">
+                        <label for="email">{% trans "Email" %}</label>
+                        <input id="email" maxlength="80" name="email" size="20" type="text">
+                    </fieldset>
+
+                    <fieldset class="form-item">
+                        <label for="00N0B000006X3fw">{% trans "Language" %}</label>
+                        <select id="00N0B000006X3fw" name="00N0B000006X3fw" title="Language">
```
Comment (Pomax) [inline donate/templates/pages/core/contributor_support_page.html]: @alanmoo or @stephaniemcv, is this value related to anything we can set? Or is this just the id that salesforce will always use for a language picker?

File: donate/templates/pages/core/contributor_support_page.html [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        <h1 class="page-header__heading">{{ page.title }}</h1>
+    </div>
+</header>
+
+<div class="layout">
+    <div class="layout__container">
+        <div class="layout__full-col">
+            <div class="page-body">
+                <p>{% blocktrans %}
+                    Here is some intro guide text to explain what this form is for and who should be filling it out.
```
Comment (alanmoo) [inline donate/templates/pages/core/contributor_support_page.html]:
```suggestion
                    If you need help with a contribution, fill out this form and we'll get back to soon.
```
Good enough for now

File: source/sass/components/_support-form.scss [hunk 8]
```diff
@@ -0,0 +1,11 @@
+.support-form {
+  width: 500px;
```
Comment (alanmoo) [inline source/sass/components/_support-form.scss]:
```suggestion
  max-width: 500px;
```
Don't break mobile

File: source/sass/components/_support-form.scss [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+
+  fieldset {
+    border: none;
+
+    label {
+      display: block;
```
Comment (alanmoo) [inline source/sass/components/_support-form.scss]: This doesn't seem to be necessary now?

File: donate/settings.py (hunk 3 above)
Comment (Pomax) [reply donate/settings.py]: yeah - @cadecairos's updating his settings PR, which also fixes this but "properly".

File: donate/settings.py (hunk 4 above)
Comment (Pomax) [reply donate/settings.py]: env var. Adding a thunderbird app override was more work for no payoff beyond what the env var does for us.

File: donate/templates/pages/core/contributor_support_page.html [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+            <div class="page-body">
+                <form class="support-form action="https://mozillasalescloud--mkl.cs4.my.salesforce.com/servlet/servlet.WebToCase?encoding=UTF-8" method="POST">
+                    <p class="rich-text">{% blocktrans %}
+                        Here is some intro guide text to explain what this form is for and who should be filling it out.
+                    {% endblocktrans %}</p>
+
+                    <input type=hidden name="orgid" value="{{ orgid }}">
+
+                    {# TODO: should the retURL be a static "thank you for your question" page rather than "the current page"? #}
+                    <input type=hidden name="retURL" value="{{ page.full_url }}">
```
Comment (Pomax) [inline donate/templates/pages/core/contributor_support_page.html]: This is an open question... @alanmoo @stephaniemcv

File: source/sass/components/_support-form.scss (hunk 9 above)
Comment (Pomax) [reply source/sass/components/_support-form.scss]: nope, removing.

File: donate/templates/pages/core/contributor_support_page.html [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+    </div>
+</header>
+
+<div class="layout">
+    <div class="layout__container">
+        <div class="layout__full-col">
+            <div class="page-body">
+                <form class="support-form action="https://mozillasalescloud--mkl.cs4.my.salesforce.com/servlet/servlet.WebToCase?encoding=UTF-8" method="POST">
+                    <p class="rich-text">{% blocktrans %}
+                        Here is some intro guide text to explain what this form is for and who should be filling it out.
```
Comment (Pomax) [inline donate/templates/pages/core/contributor_support_page.html]:
```suggestion
                        If you need help with a contribution, fill out this form and we'll get back to soon.
```

File: donate/templates/pages/core/contributor_support_page.html (hunk 10 above)
Comment (alanmoo) [reply donate/templates/pages/core/contributor_support_page.html]: I have no opinion/idea. Do whatever's easy.

File: donate/templates/pages/core/contributor_support_page.html [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+    </div>
+</header>
+
+<div class="layout">
+    <div class="layout__container">
+        <div class="layout__full-col">
+            <div class="page-body">
+                <form class="support-form action="https://mozillasalescloud--mkl.cs4.my.salesforce.com/servlet/servlet.WebToCase?encoding=UTF-8" method="POST">
+                    <p class="rich-text">{% blocktrans %}
+                        If you need help with a contribution, fill out this form and we'll get back to soon.
```
Comment (alanmoo) [inline donate/templates/pages/core/contributor_support_page.html]:
```suggestion
                        If you need help with a contribution, fill out this form and we'll get back to you soon.
```

File: source/sass/components/_support-form.scss [hunk 13]
```diff
@@ -0,0 +1,12 @@
+.support-form {
+  max-width: 500px;
```
Comment (Pomax) [inline source/sass/components/_support-form.scss]:
```suggestion
  max-width: 500px;
  min-height: 30vh;
```

**Comment by Pomax** [comment]
It looks like this is generating a form action error:
```
Refused to send form data to 'https://mozillasalescloud--mkl.cs4.my.salesforce.com/servlet/servlet.WebToCase?encoding=UTF-8' because it violates the following Content Security Policy directive: "form-action 'self'".
```
@cadecairos what would be the way to clear that up?

**Comment by cadecairos** [comment]
> @cadecairos what would be the way to clear that up?

Assuming this is a form POST action, you want to add that hostname to the CSP_FORM_ACTION variable

File: donate/templates/pages/core/contributor_support_page.html [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+                </div>
+            {% else %}
+                <form class="" action="https://mozillasalescloud--mkl.cs4.my.salesforce.com/servlet/servlet.WebToCase?encoding=UTF-8" method="POST">
+                    <p class="rich-text">{% blocktrans %}
+                        If you need help with a contribution, fill out this form and we'll get back to you soon.
+                    {% endblocktrans %}</p>
+
+                    <input type=hidden name="orgid" value="{{ orgid }}">
+                    <input type=hidden name="retURL" value="{{ page.full_url }}?submitted=true">
+
```
Comment (stephaniemcv) [inline donate/templates/pages/core/contributor_support_page.html]: Add: <input type=hidden name="recordType" id="recordType" value="012P0000000SW6LIAW">

Comment (stephaniemcv) [inline donate/templates/pages/core/contributor_support_page.html]:
```suggestion
<input type=hidden name="recordType" id="recordType" value="012P0000000SW6LIAW">
```

File: donate/templates/pages/core/contributor_support_page.html [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+                        <div class="form-item__wrapper form-custom-select">
+                            <select id="type" name="type">
+                                <option value="">--{% trans "None" %}--</option>
+                                <option value="Problem">{% trans "Problem" %}</option>
+                                <option value="Feature Request">{% trans "Feature Request" %}</option>
+                                <option value="Question">{% trans "Question" %}</option>
+                                <option value="Customer Care">{% trans "Customer Care" %}</option>
+                                <option value="Fundraising">{% trans "Fundraising" %}</option>
+                                <option value="Legal">{% trans "Legal" %}</option>
+                                <option value="Technical Issue">{% trans "Technical Issue" %}</option>
```
Comment (stephaniemcv) [inline donate/templates/pages/core/contributor_support_page.html]: Remove all options except, including "other" except for the value of "Thunderbird".

File: donate/templates/pages/core/contributor_support_page.html [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+    </div>
+</header>
+
+<div class="layout">
+    <div class="layout__container">
+        <div class="layout__full-col">
+            <div class="page-body support-form">
+            {% if 'submitted' in request.GET %}
+                <h1 class="heading heading--primary">
+                    {% trans "Thank you for contacting us." %}
```
Comment (stephaniemcv) [inline donate/templates/pages/core/contributor_support_page.html]:
```suggestion
                    {% trans "Thank you for contacting us. A case has been created for your issue and we will be in touch shortly." %}
```

File: donate/templates/pages/core/contributor_support_page.html (hunk 14 above)
Comment (Pomax) [reply donate/templates/pages/core/contributor_support_page.html]: is this is a fixed value or does it need a variable set for it?

Comment (stephaniemcv) [reply donate/templates/pages/core/contributor_support_page.html]: It's a fixed value.

File: donate/templates/pages/core/contributor_support_page.html (hunk 15 above)
Comment (Pomax) [reply donate/templates/pages/core/contributor_support_page.html]: I'm... not sure I understood that one? =D

**Comment by Pomax** [comment]
This is a universal page type, but we'll only make a help page using it on the thunderbird instance. The `contributor_support_page.html` file inherits everything that directly showin in it from other files that have TB-specific tweaks applied to them already (we review app just doesn't show those tweaks since it's been told to run as "the normal site" since for review purposes things outside the actual form isn't touched by this PR)

File: donate/templates/pages/core/contributor_support_page.html (hunk 15 above)
Comment (stephaniemcv) [reply donate/templates/pages/core/contributor_support_page.html]: I want to set the only available submitting value to 'Thunderbird' It's more of an internal classifying field for donor support than an external field. Or maybe we can hide the field?

Comment (Pomax) [reply donate/templates/pages/core/contributor_support_page.html]: yeah, let's make that a hidden field if there's only one option.
